### PR TITLE
[automate-ui] Bump package versions

### PR DESCRIPTION
**Pull request by msorens**
### :nut_and_bolt: Description: What code changed, and why?

Periodic maintenance, bringing outdated npm packages current.

Reduced outdated packages from these 47 (be sure to scroll!) ...
```
$ npm outdated
Package                               Current      Wanted       Latest  Location
@angular-devkit/build-angular         0.900.6     0.900.7      0.901.4  automate-ui
@angular/animations                     9.0.6       9.1.4        9.1.4  automate-ui
@angular/cdk                            9.1.2       9.2.2        9.2.2  automate-ui
@angular/cli                            9.0.6       9.1.4        9.1.4  automate-ui
@angular/common                         9.0.6       9.1.4        9.1.4  automate-ui
@angular/compiler                       9.0.6       9.1.4        9.1.4  automate-ui
@angular/compiler-cli                   9.0.6       9.1.4        9.1.4  automate-ui
@angular/core                           9.0.6       9.1.4        9.1.4  automate-ui
@angular/forms                          9.0.6       9.1.4        9.1.4  automate-ui
@angular/language-service               9.0.6       9.1.4        9.1.4  automate-ui
@angular/material                       9.1.2       9.2.2        9.2.2  automate-ui
@angular/platform-browser               9.0.6       9.1.4        9.1.4  automate-ui
@angular/platform-browser-dynamic       9.0.6       9.1.4        9.1.4  automate-ui
@angular/platform-server                9.0.6       9.1.4        9.1.4  automate-ui
@angular/router                         9.0.6       9.1.4        9.1.4  automate-ui
@ngrx/effects                           9.0.0       9.1.0        9.1.0  automate-ui
@ngrx/entity                            9.0.0       9.1.0        9.1.0  automate-ui
@ngrx/router-store                      9.0.0       9.1.0        9.1.0  automate-ui
@ngrx/store                             9.0.0       9.1.0        9.1.0  automate-ui
@ngrx/store-devtools                    9.0.0       9.1.0        9.1.0  automate-ui
@nguniversal/express-engine        9.0.0-rc.2  9.0.0-rc.2        9.1.0  automate-ui
@types/faker                           4.1.10      4.1.11       4.1.11  automate-ui
@types/jasmine                         3.3.10      3.3.10       3.5.10  automate-ui
@types/lodash                        4.14.149    4.14.150     4.14.150  automate-ui
@types/node                          12.12.29    12.12.37      13.13.4  automate-ui
ajv                                    6.12.0      6.12.2       6.12.2  automate-ui
codelyzer                               5.2.1       5.2.2        5.2.2  automate-ui
core-js                                 3.6.4       3.6.5        3.6.5  automate-ui
d3                                     5.15.0      5.16.0       5.16.0  automate-ui
diff2html                              2.12.2      2.12.2        3.1.6  automate-ui
hock                                    1.4.0       1.4.1        1.4.1  automate-ui
immutable                               3.8.2       3.8.2  4.0.0-rc.12  automate-ui
jasmine-spec-reporter                   4.2.1       4.2.1        5.0.2  automate-ui
karma                                   4.4.1       4.4.1        5.0.4  automate-ui
karma-coverage-istanbul-reporter        2.1.1       2.1.1        3.0.0  automate-ui
karma-jasmine-html-reporter             1.5.2       1.5.3        1.5.3  automate-ui
moment                                 2.24.0      2.25.1       2.25.1  automate-ui
node-sass                              4.13.1      4.14.0       4.14.0  automate-ui
protractor                              5.4.3       5.4.4        5.4.4  automate-ui
rrule                                   2.4.1       2.4.1        2.6.4  automate-ui
rxjs                                    6.5.4       6.5.5        6.5.5  automate-ui
rxjs-tslint                             0.1.7       0.1.8        0.1.8  automate-ui
ts-node                                 8.6.2      8.10.1       8.10.1  automate-ui
tslint                                 5.20.1      5.20.1        6.1.2  automate-ui
typescript                              3.7.5       3.7.5        3.8.3  automate-ui
uuid                                    7.0.3       7.0.3        8.0.0  automate-ui
zone.js                                0.10.2      0.10.3       0.10.3  automate-ui
```
to these seven...
```
$ npm outdated
Package                         Current      Wanted       Latest  Location
@nguniversal/express-engine  9.0.0-rc.2  9.0.0-rc.2        9.1.0  automate-ui
@types/jasmine                   3.3.10      3.3.10       3.5.10  automate-ui
@types/node                    12.12.37    12.12.37      13.13.4  automate-ui
immutable                         3.8.2       3.8.2  4.0.0-rc.12  automate-ui
rrule                             2.4.1       2.4.1        2.6.4  automate-ui
tslint                           5.20.1      5.20.1        6.1.2  automate-ui
typescript                        3.7.5       3.7.5        3.8.3  automate-ui
```

All of the seven packages above are explicitly version-locked in package.json. As you cannot comment a JSON file, the reasons are enumerated in automate-ui's top-level README.md file.

This PR would have been just 2 files (package.json and package-lock.json) but for a peculiar breaking change in `moment`, requiring applying a workaround in 57 files. (As workarounds go, though, it is rather clean.)

Also included in this PR are fixes for lint warnings, which do not fail the build so they crept in recently. (I will likely do a follow-up to activate treat-warnings-as-errors for lint warnings as I did recently for unit test warnings.)

This PR also fixes almost all security flaws reported by `npm audit`. There are just 4 remaining of low severity that cannot be auto-fixed.

### :chains: Related Resources

The last run was done in PR #3057.

### :+1: Definition of Done

1. `npm install` reports no changes to package.json or package-lock.json.

2. The following have no warnings or errors:
 - `ng build`
 - `make unit`
 - `make lint`

3. The app runs in the browser with no console errors.

### :athletic_shoe: How to Build and Test the Change

Do your usual hab studio setup, notably including building `automate-ui-devproxy`. 

Then (line numbers shown for reference):
```
[1] cd components/automate-ui
[2] mv node_modules /location/to/save/your/old/node_modules
[3] npm install
[4] make serve
```

Notes:
Line [2] is only needed if you want to be able to easily resume what you were doing before you tried out this PR! Once you are done evaluating, delete your new `node_modules` and restore your copy made in line [2].
Line [3] generates the new `node_modules` directory for this PR. Important!  `git status` should report no changes to package-lock.json after this.
Line [4] starts automate-ui.

### :white_check_mark: Checklist

- [x] I have read the [CONTRIBUTING document](https://github.com/chef/automate/blob/master/CONTRIBUTING.md).
- [x] Tests added/updated?
- [ ] Docs added/updated?
- [x] All commits have been signed-off for the [Developer Certification of Origin](https://github.com/chef/chef/blob/master/CONTRIBUTING.md#developer-certification-of-origin-dco).

### CODE REVIEW

**Comment by susanev** [comment]
can you collaborate with @tarablack01 on this if you haven't already? they are trying to get the carbon design system added and we should confirm that these additional updates wont be more of a blocker for that work.

File: components/automate-ui/src/app/components/calendar/calendar.component.ts [hunk 1]
```diff
@@ -3,7 +3,7 @@ import { Component,
          Output,
          EventEmitter,
          ChangeDetectionStrategy } from '@angular/core';
-import * as m from 'moment';
+import * as moment from 'moment/moment';
```
Comment (msorens) [inline components/automate-ui/src/app/components/calendar/calendar.component.ts:6]: This is the workaround for `moment` mentioned in the preamble. This one particular file was coincidentally unique by starting with `import * as m from...` whereas every other file had the canonical `import * as moment from...` so this file changes to that standard, too.

File: components/automate-ui/src/app/modules/infra-proxy/infra-role-details/infra-role-details.component.spec.ts [hunk 2]
```diff
@@ -102,7 +102,7 @@ describe('InfraRoleDetailsComponent', () => {
     component = fixture.componentInstance;
     element = fixture.debugElement;
     fixture.detectChanges();
-    router = TestBed.get(Router);
+    router = TestBed.inject(Router);
```
Comment (msorens) [inline components/automate-ui/src/app/modules/infra-proxy/infra-role-details/infra-role-details.component.spec.ts:105]: This and the next file generated lint warnings due to method deprecation, so fixed those.

File: components/automate-ui/README.md [hunk 3]
```diff
@@ -74,34 +74,23 @@ error TS6200: Definitions of the following identifiers conflict with those in an
...
 
-**Package @types/node: ~12.12.29**
+### Package @types/node: ~12.12.29
 
 Reason: The major version should match our node version, which is 12.13.1, so I have constrained it to the highest version less than that.
 That is a guess on my part, could not really confirm what version one should be running, through these references:
- - https://www.npmjs.com/package/@types/node
- - https://github.com/DefinitelyTyped/DefinitelyTyped#how-do-definitely-typed-package-versions-relate-to-versions-of-the-corresponding-library
 
-**Package diff2html: ^2.12.2**
```
Comment (msorens) [inline components/automate-ui/README.md:84]: I patched breaking changes from this package and unlocked the version. However, need someone to actually test this in the UI; I could not quite tell where it shows up.

Comment (msorens) [reply components/automate-ui/README.md:84]: @Shadae and/or @scottopherson: this file has your fingerprints, so when you have a chance, please check (or tell me how to)>

**Comment by msorens** [comment]
FYI: Would like to hold merging until getting a confirmation on the Diff2Html changes.

File: components/automate-ui/README.md (hunk 3 above)
Comment (scottopherson) [reply components/automate-ui/README.md:84]: @msorens the only place I see `diff2html` being used is in `app-delta-viewer` which is only used by `app-resource-item`. Everything seems to look fine on the client-runs node details page which has the only usage of `app-resource-item`:

Screen Shot 2020-05-05 at 12 05 05 PM

(though the padding within the diff rows seems a bit much imo)

**Pull request merged**